central/notifiers/metadatagetter/datastore_impl.go

```diff
@@ -88,8 +88,12 @@ func getNamespaceFromAlert(ctx context.Context, alert *storage.Alert, namespaceS
 		return nil
 	}
 
-	if namespaceName == "" || clusterID == "" {
-		log.Errorf("Alert entity doesn't contain namespace and cluster ID: %+v", alert.GetEntity())
+	if namespaceName == "" {
+		return nil
```

Comment:
For a certain kind of alerts, `namespace` is expected. I would recommend logging errors for those cases.